assets/app/actions/siteActions.js

```diff
@@ -65,7 +66,15 @@ export default {
 
   fetchSiteConfigsAndAssets(site) {
     return this.fetchSiteConfigs(site).then((site) => {
-      return this.fetchSiteAssets(site);
+      return this.fetchSiteAssets(site).then((site) => {
```

Comment:
I'm wondering then if we shouldn't try to enforce a navigation.json to be in every repo, since the extra request is basically only required to develop locally.